objectivec/GPBWellKnownTypes.m

```diff
@@ -115,3 +115,52 @@ - (void)setTimeIntervalSince1970:(NSTimeInterval)timeIntervalSince1970 {
 }
 
 @end
+
+NSString *const GPBTypeGoogleApisComPrefix = @"type.googleapis.com/";
+
+@implementation GPBAny (GBPWellKnownTypes)
+
+- (NSString*)typeName {
+  NSAssert([self.typeURL hasPrefix:GPBTypeGoogleApisComPrefix],
+           @"Invalid any type url (%@).", self.typeURL);
+  if (![self.typeURL hasPrefix:GPBTypeGoogleApisComPrefix]) {
+    return @"";
```

Comment:
If we return @"", we can treat it as normal return value without additional checking for nil.
In the places calling typeName, we can guarantee that returned value from typeName is always compared with a non-empty string.